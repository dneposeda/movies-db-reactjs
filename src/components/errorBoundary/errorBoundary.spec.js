import React from 'react';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
    it('renders correctly', () => {
        const component = shallow (
            <ErrorBoundary />
        );
        expect(component.state('hasError')).toEqual(false);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly Error', () => {
        const component = shallow (
            <ErrorBoundary />
        );
        component.setState({ hasError: true });
        expect(component.state('hasError')).toEqual(true);
        expect(component).toMatchSnapshot();
        expect(component.text()).toMatch('Что-то пошло не так.');
    });
});
