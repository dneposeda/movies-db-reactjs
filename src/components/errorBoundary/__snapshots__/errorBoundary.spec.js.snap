// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders correctly 1`] = `""`;

exports[`ErrorBoundary renders correctly Error 1`] = `
<h1>
  Что-то пошло не так.
</h1>
`;
